modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -2995,6 +2995,12 @@ private RestoreLogicalState applyLogicalUpdates(
                         GridCacheContext cacheCtx = cctx.cacheContext(cacheId);
 
                         if (cacheCtx != null) {
+                            // Index was destroyed before restart, mark it as invalid, so that we don't start
+                            // hew destruction task. Existance of this record means that
+                            // DurableBackgroundTask that destroys indexes is already in metastorage and
+                            // that it will destroy index.
+                            cacheCtx.kernalContext().query().getIndexing().invalidateIndex(cacheCtx, record.oldTreeName());
```

Comment:
How long is this line? Maybe it would be convenient to extract `...query()` into a variable, but it's up to you